packit_service/worker/handlers/distgit.py

```diff
@@ -408,33 +409,58 @@ def __init__(
+        if self.data.event_type in (
+            PushPagureEvent.__name__,
+            PullRequestMergedPagureEvent.__name__,
+        ):
+            if self.dg_branch not in (
                 configured_branches := get_branches(
                     *self.job_config.dist_git_branches,
                     default="main",
                     with_aliases=True,
                 )
             ):
                 logger.info(
-                    f"Skipping build on '{self.data.git_ref}'. "
+                    f"Skipping build on '{self.dg_branch}'. "
                     f"Koji build configured only for '{configured_branches}'."
                 )
                 return False
 
-            # Packit should only build its own contributions
-            # We need to preserve the proven packager workflow - pushes not done by Packit
-            # should be built by those that do it, e.g. rebuilds in a side tag.
-            # prod/stg filtering happens in jobs.py -> packit_instances
-            if not (
-                self.data.event_dict["email"] == "hello@packit.dev"
-                and self.data.event_dict["name"] == "Packit"
+        if self.data.event_type == PushPagureEvent.__name__:
+            if self.data.event_dict["committer"] == "pagure":
+                logger.info(
+                    f"Push event {self.data.identifier} as a result of merged pull request."
+                    f"We will handle this by reacting to the closed PR message."
+                )
+                return False
+
+            if (
+                self.data.event_dict["committer"]
+                not in self.job_config.allowed_committers
             ):
                 logger.info(
-                    f"Push event {self.data.identifier} ({self.data.event_dict['name']} "
-                    f"<{self.data.event_dict['email']}>) not done by us."
+                    f"Push event {self.data.identifier} done by "
+                    f"{self.data.event_dict['committer']} that is not allowed in project "
+                    f"configuration: {self.job_config.allowed_committers}."
                 )
                 return False
+
+        if self.data.event_type == PullRequestMergedPagureEvent.__name__:
+            if (
+                self.data.event_dict["committer"]
+                not in self.job_config.allowed_committers
+                and self.data.event_dict["pr_author"]
+                not in self.job_config.allowed_pr_authors
+            ):
+                logger.info(
+                    f"Pull request merged event {self.data.identifier} for PR created by"
+                    f" {self.data.event_dict['pr_author']} that is not allowed in project "
+                    f"configuration: {self.job_config.allowed_pr_authors} and merged by"
+                    f" {self.data.event_dict['committer']} that is also not allowed in "
+                    f"project configuration: {self.job_config.allowed_committers}."
+                )
```

Comment:
Can we add something to make it clear that one of these conditions needs to be fulfilled?
Something like this?

```suggestion
                logger.info(
                    f"Pull request merged event {self.data.identifier} for PR created by"
                    f" {self.data.event_dict['pr_author']} that is not allowed in project "
                    f"configuration: {self.job_config.allowed_pr_authors} and merged by"
                    f" {self.data.event_dict['committer']} that is also not allowed in "
                    f"project configuration: {self.job_config.allowed_committers}. "
                    "One of these conditions needs to be fulfilled."
                )
```